aws/resource_aws_db_proxy.go

```diff
@@ -0,0 +1,413 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsDbProxy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDbProxyCreate,
+		Read:   resourceAwsDbProxyRead,
+		Update: resourceAwsDbProxyUpdate,
```

Comment:
New resources that implement update in-place support should include acceptance testing to verify the functionality for each update-able attribute. For example,

```go
func TestAccAWSDBProxy_DebugLogging(t *testing.T) {
  var dbProxy rds.DBProxy
  resourceName := "aws_db_proxy.test"
  rName := acctest.RandomWithPrefix("tf-acc-test")

  resource.ParallelTest(t, resource.TestCase{
    PreCheck:     func() { testAccPreCheck(t) },
    Providers:    testAccProviders,
    CheckDestroy: testAccCheckAWSDBProxyDestroy,
    Steps: []resource.TestStep{
      {
        Config: testAccAWSDBProxyConfigDebugLogging(name, true),
        Check: resource.ComposeTestCheckFunc(
          testAccCheckAWSDBProxyExists(resourceName, &dbProxy),
          resource.TestCheckResourceAttr(resourceName, "debug_logging", "true"),
        ),
      },
      {
        ResourceName:      resourceName,
        ImportState:       true,
        ImportStateVerify: true,
      },
      {
        Config: testAccAWSDBProxyConfigDebugLogging(name, false),
        Check: resource.ComposeTestCheckFunc(
          testAccCheckAWSDBProxyExists(resourceName, &dbProxy),
          resource.TestCheckResourceAttr(resourceName, "debug_logging", "false"),
        ),
      },
    },
  })
}

func TestAccAWSDBProxy_IdleClientTimeout(t *testing.T) {
  var dbProxy rds.DBProxy
  resourceName := "aws_db_proxy.test"
  rName := acctest.RandomWithPrefix("tf-acc-test")

  resource.ParallelTest(t, resource.TestCase{
    PreCheck:     func() { testAccPreCheck(t) },
    Providers:    testAccProviders,
    CheckDestroy: testAccCheckAWSDBProxyDestroy,
    Steps: []resource.TestStep{
      {
        Config: testAccAWSDBProxyConfigIdleClientTimeout(name, 900),
        Check: resource.ComposeTestCheckFunc(
          testAccCheckAWSDBProxyExists(resourceName, &dbProxy),
          resource.TestCheckResourceAttr(resourceName, "idle_client_timeout", "900"),
        ),
      },
      {
        ResourceName:      resourceName,
        ImportState:       true,
        ImportStateVerify: true,
      },
      {
        Config: testAccAWSDBProxyConfigIdleClientTimeout(name, 3600),
        Check: resource.ComposeTestCheckFunc(
          testAccCheckAWSDBProxyExists(resourceName, &dbProxy),
          resource.TestCheckResourceAttr(resourceName, "idle_client_timeout", "3600"),
        ),
      },
    },
  })
}

func testAccAWSDBProxyConfigDebugLogging(rName string, debugLogging bool) string {
  return fmt.Sprintf(`
# ... other configuration ...

resource "aws_db_proxy" "test" {
  # ... other configuration  ...

  debug_logging = %[2]t
}
`, rName, debugLogging)
}

func testAccAWSDBProxyConfigIdleClientTimeout(rName string, idleClientTimeout int) string {
  return fmt.Sprintf(`
# ... other configuration ...

resource "aws_db_proxy" "test" {
  # ... other configuration  ...

  idle_client_timeout = %[2]d
}
`, rName, idleClientTimeout)
}
```

Especially for configurations that require a lot of other resource setup, its typically easiest to create a "base" configuration that the other test configurations can include. 👍

Reply:
Still pending 👍